File: 04_TSBackend/server/src/utils/sendSMS.ts
import twilio from "twilio"
import config from "config"


const SID: string = config.get<string>("SID")
const TOKEN: string = config.get<string>("TOKEN")
const PHONE: string = config.get<string>("PHONE")

interface SmsData {
    body : string,
    to: string
}

let client = new twilio.Twilio(SID, TOKEN)


async function SendSMS(smsData: SmsData) {
    try {
 
        await client.messages.create({
            body: smsData.body,
            to: smsData.to,
            from: PHONE
        })

        console.log(`sms sent✅`);
        

        
    } catch (error) {
        console.log(error);
        
    }
    
}

export default SendSMS;

File: 08_Go_Gin_PostgreSQL/server/internal/utils/sendSMS.go
package utils

import (
	"log"

	"github.com/FullStackWebDev_2026/08_Go_Gin_PostgreSQL/server/internal/config"
	"github.com/twilio/twilio-go"
	openapi "github.com/twilio/twilio-go/rest/api/v2010"
)

type SMSData struct {
	To   string
	Body string
}

func SendSMS(smsData SMSData) error {
	// create client
	client := twilio.NewRestClientWithParams(twilio.ClientParams{
		Username: config.AppConfig.SID,
		Password: config.AppConfig.TOKEN,
	})

	// send sms
	_, err := client.Api.CreateMessage(&openapi.CreateMessageParams{
		From: &config.AppConfig.PHONE,
		To:   &smsData.To,
		Body: &smsData.Body,
	})

	if err != nil {
		log.Printf("couldn't send sms")
		return err
	}

	log.Printf("Sms sent✅")
	return nil
}
